### PR TITLE
Deprecate collective link fields

### PR DESCRIPTION
**Pull request by hdiniz**
Related https://github.com/opencollective/opencollective/issues/6284.

Deprecate link properties in favor of generic social links

### CODE REVIEW

**Comment by znarf** [comment]
Looks good to me so far. What's missing?

**Comment by hdiniz** [comment]
> Looks good to me so far. What's missing?

I wanted to add a change to update the social links when the deprecated fields are used. Can you make another pass with the latest commit?

File: server/models/Collective.js [hunk 1]
```diff
@@ -912,6 +921,45 @@ Collective.prototype.updateSocialLinks = async function (socialLinks) {
...
+    }
+
+    const websiteSocialLink = updatedSocialLinks.find(sl => sl.type === SocialLinkType.WEBSITE);
+    if (websiteSocialLink && websiteSocialLink.url) {
+      collectiveFields.website = websiteSocialLink.url;
+    }
+
+    const repositorySocialLink = updatedSocialLinks.find(sl => {
+      return sl.type === SocialLinkType.GIT || sl.type === SocialLinkType.GITHUB || sl.type === SocialLinkType.GITLAB;
+    });
```
Comment (znarf) [inline server/models/Collective.js:948]:
```suggestion
    const repositorySocialLink = updatedSocialLinks.find(sl => [SocialLinkType.GIT, SocialLinkType.GITHUB, SocialLinkType.GITLAB].includes(sl.type));
```

Comment (znarf) [reply server/models/Collective.js:948]: Just a suggestion.

**Review by znarf — APPROVED** [review]
Looks good!

**Pull request merged**